aqua-ethos-setup/aqua_ethos.sh

```diff
@@ -85,128 +70,88 @@ function login {
-	fi
-fi
+function replaceConfigs {
+	log "Replacing ETH configs in $CONFIG_FILE"
+
+	# Note: using "@" instead of "/" as delimiter because some expressions contain slashes (URLs)
+	sed -i.bak "s@ETH_ARTIFACTORY_URL@$ARTIFACTORY_URL@g" "$CONFIG_FILE"
+	sed -i.bak "s@ETH_ARTIFACTORY_PREFIX@$ARTIFACTORY_PREFIX@g" "$CONFIG_FILE"
+	sed -i.bak "s@ETH_ARTIFACTORY_USERNAME@$ARTIFACTORY_USERNAME@g" "$CONFIG_FILE"
+	sed -i.bak "s@ETH_ARTIFACTORY_PASSWORD@$ARTIFACTORY_PASSWORD@g" "$CONFIG_FILE"
+	sed -i.bak "s@ETH_QUALYS_USERNAME@$QUALYS_USERNAME@g" "$CONFIG_FILE"
+	sed -i.bak "s@ETH_QUALYS_PASSWORD@$QUALYS_PASSWORD@g" "$CONFIG_FILE"
+	#sed -i.bak "s@ETH_ENCRYPT_ENV_VARS@$ENCRYPT_ENV_VARS@g" "$CONFIG_FILE"
+
+	# Update the encryption mode
+	# cat $CONFIG_FILE | jq -r '. | select(policies.security_profiles[].name=="Ethos") | .encrypt_all_envs |= '$ENCRYPT_ENV_VARS''
+	cat $CONFIG_FILE | jq '.policies.security_profiles[0].encrypt_all_envs = '$ENCRYPT_ENV_VARS'' > $CONFIG_FILE.bak
+	mv $CONFIG_FILE.bak $CONFIG_FILE
+
+	# Empty out the images array in case it already exists
+	log "Clearing the old images array"
+	cat $CONFIG_FILE | jq '.images |= []' > $CONFIG_FILE.bak
+	mv $CONFIG_FILE.bak $CONFIG_FILE
+
+	log "Adding approved images: $APPROVED_IMAGES"
+	IFS=',' read -ra ADDR <<< "$APPROVED_IMAGES"
+	for IMAGE in "${ADDR[@]}"; do
+		if [[ $EXISTING_IMAGES == *"$IMAGE"* ]]; then
+			log "Image already exists. Skipping: $IMAGE"
+			continue;
+		fi
+
+		REPO=$(echo $IMAGE | cut -d':' -f1)
+
+	    cat $CONFIG_FILE | jq '.images |= .+ [{"Name":"'$IMAGE'","Repository":"'$REPO'","PolicyName":"","Registry":"artifactory-admin","Labels":["production approved"]}]' > $CONFIG_FILE.bak
+	    mv $CONFIG_FILE.bak $CONFIG_FILE
+	done
+}
 
-if [[ ! -d $CRED_DIR/configs ]]; then
-    sudo mkdir $CRED_DIR/configs -p
-fi
+setup
+waitForWeb
+login
+getExistingImages
+replaceConfigs
 
-sudo chmod 0755 $CRED_DIR/configs
-sudo chown -R $(whoami):$(whoami) $CRED_DIR/configs
+# Import the config file
+curl --silent -H "Content-Type: application/json" -H "$HEADER: Bearer $TOKEN" -X POST -d "@$CONFIG_FILE" "$WEB_URL/settings/import"
 
-# Obtain the runtime policy and set as ethos default
-curl --silent -H "$HEADER: Bearer $TOKEN" -X GET $WEB_URL/runtime_policy > $CRED_DIR/configs/threat1_mitigation.json
-sudo chmod 0755 $CRED_DIR/configs/threat1_mitigation.json
-sudo cat $CRED_DIR/configs/threat1_mitigation.json | jq --arg default_security_profile Ethos '. + {default_security_profile: $default_security_profile}' > $CRED_DIR/configs/threat_mitigation.json
-sudo chmod 0755 $CRED_DIR/configs/threat_mitigation.json
-curl --silent -H "$HEADER: Bearer $TOKEN" -X PUT -d @$CRED_DIR/configs/threat_mitigation.json $WEB_URL/runtime_policy
-sudo rm -rf $CRED_DIR/configs
 
 # HEALTHCHECK
 function healthcheck {
-	if [[ $(expr $(date +%s) - $(date +%s -r $CRED_DIR/login)) -gt 1800 ]]; then
+	if [[ $(expr $(date +%s) - $(date +%s -r $DIR/login)) -gt 1800 ]]; then
```

Comment:
I just checked and $DIR isn't an internal variable, but might be a good idea to keep the name unique to what we're doing just in case ...

http://www.tldp.org/LDP/abs/html/internalvariables.html